lom/src/engine/engine.go

```diff
@@ -0,0 +1,223 @@
+     * and the main loop invokes sequence handler.
+     *
+     * This way any logical processing by sequence handler is via the context
+     * of single Go routine only.
+     */
+    chSeqHandler := make(chan int64, 2)
+    InitSeqHandler(chSeqHandler)
+
+    server := GetServerReqHandler()
+
+    /* Intimate the start of loop */
+    p.chTrack <- 0
+loop:
+    for {
+        select {
+        case msg := <- p.chClientReq:
+            server.processRequest(msg)
+
+        case <- chSeqHandler:
+            GetSeqHandler().processTimeout()
+
+        case sig := <- chSignal:
+            switch(sig) {
+            case syscall.SIGHUP, syscall.SIGINT:
+                /*
+                 * Reload.
+                 * NOTE: Any currently active sequence will not be affected
+                 * On any error, continues to use last loaded values. 
+                 */
+                 InitConfigMgr(cfgFiles)
+
+            case syscall.SIGTERM:
+                break loop
+            }
+        case <- p.chTerminate:
+            break loop
+        }
+    }
+} 
+
+func (p *engine_t) close() {
+    if len(p.chTerminate) < cap(p.chTerminate) {
+        p.chTerminate <- 1
+    }
+}
+
+func startUp(progname string, args []string) (*engine_t, error) {
+
+    path := ""
+    {
+        p := ""
+        flags := flag.NewFlagSet(progname, flag.ContinueOnError)
+        var buf bytes.Buffer
+        flags.SetOutput(&buf)
+
+        flags.StringVar(&p, "path", "", "Config files path")
+
+        err := flags.Parse(args)
+        if  err != nil {
+            return nil, LogError("Failed to parse (%v); details(%s)", args, buf.String())
+        }
+        path = p
+    }
+
+    if len(path) == 0 {
+        if p, err := os.Getwd(); err != nil {
+            return nil, LogError("Failed to get current working dir (%v)", err)
+        } else {
+            path = p
+        }
+    }
+
+    cfgFiles = &ConfigFiles_t {
+        GlobalFl: filepath.Join(path, "globals.conf.json"),
+        ActionsFl: filepath.Join(path, "actions.conf.json"),
+        BindingsFl: filepath.Join(path, "bindings.conf.json"),
+    }
+
+    if _, err := InitConfigMgr(cfgFiles); err != nil {
+        return nil, LogError("Failed to read config; (%v)", *cfgFiles)
+    }
+   
+    InitRegistrations()
+    tx, err := ServerInit()
+    if err != nil {
+        return nil, LogError("Failed to call ServerInit")
+    }
+
+    chTrack := make(chan int, 2)     /* To track start/end of loop */
+    chTerminate := make(chan int, 1) /* To force terminate a loop */
+    engine := &engine_t{tx: tx, chTrack:chTrack, chTerminate: chTerminate }
+    go engine.runLoop()
+    
+    /* Wait for loop start */
+    <- chTrack
+    return engine, nil
+}
+
+func main() {
+
+    engine, err := startUp(os.Args[0], os.Args[1:])
+    if err != nil {
+        LogError("Engine aborting ...")
+        return
+    }
+
+    <- engine.chTrack
+    LogDebug("Loop ended")
+
+
+    LogInfo("Engine exiting...")
```

Comment:
Is it better to log as notice ?

Reply:
The exit happens only on SIGTERM, which is sent on service shutdown. 
So it does not matter.

Reply:
When it restarts will the logs are written to same file ? In that case it may be helpful to see when it finally went down

Reply:
SONiC runs at info level by default.
Yes, all logs goes to /var/log/syslog.